internal/server/server_test.go

```diff
@@ -40,6 +41,8 @@ func setupServer(t *testing.T, ops ...func(*testing.T, *Options)) *Server {
 	err = s.loadConfig(s.options.Config)
 	assert.NilError(t, err)
 
+	data.InvalidateCache()
```

Comment:
This should be part of cleanup? So that we can use the cache while the test runs, but we clean it up when the test ends?

```suggestion
	t.Cleanup(data.InvalidateCache)
```

Or maybe we want it to do both (clear in initially in case some other test did not, and also using `t.Cleanup` to clean up properly after itself) ?

Reply:
Calling this in the setup is great to ensure the state is as expected before the test runs. Calling is cleanup is fine if all other tests are well behaving but leaves more room for errors from side effects of running other tests. In this case, I'll add `InvalidateCache` to both setup and cleanup